vm/systemSmartContracts/auction.go

```diff
@@ -976,17 +988,36 @@ func (s *stakingAuctionSC) unBond(args *vmcommon.ContractCallInput) vmcommon.Ret
 			unBondValue.Set(auctionConfig.NodePrice)
 		}
 
-		_ = totalUnBond.Add(totalUnBond, unBondValue)
 		unBondedKeys = append(unBondedKeys, blsKey)
+		if nodeData.Waiting {
+			totalWaiting.Add(totalWaiting, unBondValue)
+			if registrationData.NumWaiting > 0 {
+				registrationData.NumWaiting -= 1
+			}
+			continue
+		}
+		totalUnBond.Add(totalUnBond, unBondValue)
 	}
 
 	if registrationData.LockedStake.Cmp(totalUnBond) < 0 {
-		totalUnBond.Set(registrationData.LockedStake)
+		s.eei.AddReturnMessage("contract error on unBond function")
```

Comment:
Maybe add an additional info to know that this is because totalUnBond < 0

Reply:
done